snowcap/api/protobuf/snowcap/widget/v1/widget.proto

```diff
@@ -290,6 +291,90 @@ message Image {
   }
 }
 
+message MouseArea {
+  WidgetDef child = 1;
+  bool on_press = 2;
+  bool on_release = 3;
+  bool on_double_click = 4;
+  bool on_right_press = 5;
+  bool on_right_release = 6;
+  bool on_middle_press = 7;
+  bool on_middle_release = 8;
+  bool on_scroll = 9;
+  bool on_enter = 10;
+  bool on_move = 11;
+  bool on_exit = 12;
+  optional Interaction interaction = 13;
+  optional string unique_id = 14;
+  optional uint32 widget_id = 15;
```

Comment:
I feel like these two could be merged. There are other iced widget that might need a string as an id (e.g. `TextArea` and `TextEditor`). 

I went into more depth in https://github.com/pinnacle-comp/pinnacle/pull/364#issuecomment-3330665771 as to why this is needed at least in `MouseArea`, but the gist of it is that without a way to identify this widget accross `view()` generations, the double click event message tied to the `widget_id` is dropped because a new view is created between the second `press` and the `double_click` handling in the config (the two event are sent at the same time).